scripts/pylint_extensions.py

```diff
@@ -475,19 +460,13 @@ def visit_yield(self, node):
                 method definition in the AST.
         """
         func_node = node.frame()
-        if not isinstance(func_node, astroid.FunctionDef):
-            return
 
         doc = docstrings_checker.docstringify(func_node.doc)
         if not doc.is_valid() and self.config.accept_no_yields_doc:
             return
 
-        if doc.supports_yields:
```

Comment:
doc.supports_yields will always be True in case of GoogleDocstrings. See here: [link](https://github.com/PyCQA/pylint/blob/b1ee3854ab89786bdfb53b062ffc3e56080ec4a2/pylint/extensions/_check_docs_utils.py#L524). And the docstrings in our codebase are first converted to GoogleDocstrings by docstringify method, hence the else condition is unreachable

Reply:
You do know that we do not exactly use Google Docstrings format, right? We have modified 
 the regexp according to our use case. Please look into this.

Reply:
I found that the docstring type is always of GoogleDocstrings after we call docstringify on a docstring which we are doing [here](https://github.com/oppia/oppia/blob/ea2a0db572c96c65543bf6faccebf0dde033c1e9/scripts/pylint_extensions.py#L481). Hence `supports_yields` will be True always

Reply:
Okay, in that case, this sounds good! Thanks.

Reply:
Hold on, but you can still test this by using some other format for docstrings instead of deleting it, right?

Reply:
No we are forcing every docstring to be of type GoogleDocstring here:
https://github.com/oppia/oppia/blob/ea2a0db572c96c65543bf6faccebf0dde033c1e9/scripts/docstrings_checker.py#L170

Reply:
I see, thanks!